src/lib/ui/thermostat/TemperatureControl.tsx

```diff
@@ -0,0 +1,126 @@
+  temperature: number
+  onChange: (temperature: number) => void
+  min?: number
+  max?: number
+  temperatureUnit?: 'fahrenheit' | 'celsius'
+}
+
+export function TemperatureControl({
+  variant,
+  temperature,
+  onChange,
+  min = DEFAULT_MIN,
+  max = DEFAULT_MAX,
+  temperatureUnit = 'fahrenheit',
+}: TemperatureControlProps): JSX.Element {
+  const increment = (): void => {
+    onChange(Math.min(temperature + 1, max))
+  }
+
+  const decrement = (): void => {
+    onChange(Math.max(temperature - 1, min))
+  }
+
+  return (
+    <div className='seam-temperature-control'>
+      <button className='seam-temperature-stepper-button' onClick={decrement}>
+        <TemperatureSubtractIcon />
+      </button>
+      <RangeSlider
+        variant={variant}
+        min={min}
+        max={max}
+        temperature={temperature}
+        onChange={(temperature) => {
+          onChange(temperature)
+        }}
+        temperatureUnit={temperatureUnit}
+      />
+      <button className='seam-temperature-stepper-button' onClick={increment}>
+        <TemperatureAddIcon />
+      </button>
+    </div>
+  )
+}
+
+interface RangeSliderProps {
+  variant: 'heat' | 'cool'
+  temperature: number
+  onChange: (temperature: number) => void
+  min: number
+  max: number
+  temperatureUnit: 'fahrenheit' | 'celsius'
+}
+
+function RangeSlider({
+  variant,
+  temperature,
+  onChange,
+  min,
+  max,
+  temperatureUnit,
+}: RangeSliderProps): JSX.Element {
+  const wrapRef = useRef<HTMLDivElement>(null)
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  useEffect(() => {
+    if (wrapRef.current != null && inputRef.current != null) {
+      const wrap = wrapRef.current
+      const input = inputRef.current
+
+      wrap.style.setProperty('--temperature-current', input.value)
+      wrap.style.setProperty(
+        '--temperature-min',
+        input.min === '' ? String(DEFAULT_MIN) : input.min
+      )
+      wrap.style.setProperty(
+        '--temperature-max',
+        input.max === '' ? String(DEFAULT_MAX) : input.max
+      )
+    }
+  }, [temperature])
+
+  const handleRangeChange: ChangeEventHandler<HTMLInputElement> = (
+    event
+  ): void => {
+    const wrap = wrapRef.current
+
+    if (wrap != null) {
+      const temperature = event.target.value
+      wrap.style.setProperty('--temperature-current', temperature)
+      onChange(Number(temperature))
+    }
+  }
+
+  return (
+    <div ref={wrapRef} className='seam-temperature-range-wrap'>
+      <input
+        ref={inputRef}
+        type='range'
+        min={min}
+        max={max}
+        value={temperature}
+        onChange={handleRangeChange}
+        className='seam-temperature-range'
+        data-variant={variant}
+      />
+
+      <div className='seam-floating-temperature'>
+        <span className='seam-floating-temperature-value'>{temperature}</span>
+        <span className='seam-floating-temperature-unit'>
+          º{temperatureUnit[0]?.toUpperCase()}
```

Comment:
Clever, but can we do this explicitly instead as here so we have one kind of logic for this? https://github.com/seamapi/react/blob/1c2b2d40197a2c8dce469fad9f666f80fa1335d1/src/lib/ui/thermostat/ClimateSettingStatus.tsx#L70